### PR TITLE
Feature connect security with database

### PR DESCRIPTION
**Pull request by ThePapiok**
tested with postman

### CODE REVIEW

File: src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  @Override
+  protected void doFilterInternal(
+      HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+      throws ServletException, IOException {
+    final String authHeader = request.getHeader("Authorization");
+    final String jwt;
+    final String userEmail;
+
+    if (StringUtils.isEmpty(authHeader)
+        || !org.apache.commons.lang3.StringUtils.startsWith(authHeader, "Bearer ")) {
```
Comment (kacper-dabrowski) [inline src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java:36]: `!org.apache.commons.lang3.` could be removed and we could use import instead, right?

Comment (kacper-dabrowski) [inline src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java:36]: `!org.apache.commons.lang3.` could be removed and we could use import instead, right?

File: src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      filterChain.doFilter(request, response);
+      return;
+    }
+    jwt = authHeader.substring(7);
```
Comment (kacper-dabrowski) [inline src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java:40]: what is the magic 7 here?

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+  @InjectMocks private AuthenticationServiceImpl authenticationService;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.openMocks(this);
+  }
+
+  @Test
+  public void shouldSuccesWhenGetUserTest() {
```
Comment (kacper-dabrowski) [inline src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java:45]: `succes` -> `success`

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    UserDTO userDTO = new UserDTO();
+    user.setMail(mail);
+    userDTO.setMail(mail);
+    Mockito.when(userRepository.findByMail(mail)).thenReturn(Optional.of(user));
+    Mockito.when(userConverter.createDTO(user)).thenReturn(userDTO);
+    Assertions.assertEquals(authenticationService.getUser(mail), userDTO);
+  }
+
+  @Test
+  public void shouldNullWhenUserNotFoundTest() {
```
Comment (kacper-dabrowski) [inline src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java:57]: what does null mean here? We could use more observability focused test description here

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    String mail = "test@test.com";
+    User user = new User();
+    user.setMail(mail);
+    Mockito.when(userRepository.findByMail(mail)).thenReturn(Optional.empty());
+    Assertions.assertNull(authenticationService.getUser(mail));
+  }
+
+  @Test
+  public void shouldTrueWhenRegisterUserTest() {
```
Comment (kacper-dabrowski) [inline src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java:66]: same as above, what does True mean here?

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    String mail = "test@test.com";
+    UserDTO userDTO = new UserDTO();
+    userDTO.setMail(mail);
+    Mockito.when(userRepository.save(any(User.class))).thenReturn(null);
+    Assertions.assertTrue(authenticationService.register(userDTO));
+  }
+
+  @Test
+  public void shouldFalseWhenRegisterUserTest() {
```
Comment (kacper-dabrowski) [inline src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java:75]: same as above, what does false mean here?

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+    userDTO.setPassword("Testt!123");
+    Authentication authentication = Mockito.mock(Authentication.class);
+    Mockito.when(userRepository.findByMail(userDTO.getMail())).thenReturn(Optional.of(new User()));
+    Mockito.when(authenticationManager.authenticate(any(UsernamePasswordAuthenticationToken.class)))
+        .thenReturn(authentication);
+    Assertions.assertNotNull(authenticationService.login(userDTO));
+  }
+
+  @Test
+  public void shouldNullWhenUserLoginTest() {
```
Comment (kacper-dabrowski) [inline src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java:96]: same here

File: src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java (hunk 2 above)
Comment (ThePapiok) [reply src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java:40]: if authHeader starts with "Bearer " we need to extract only JWT token and "Bearer " has 7 length, i will add variable to make it more clear

File: src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java (hunk 1 above)
Comment (ThePapiok) [reply src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java]: yes, i changed

Comment (ThePapiok) [reply src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java]: yes, i changed

File: src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java (hunk 2 above)
Comment (ThePapiok) [reply src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java]: i added variable

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java (hunk 3 above)
Comment (ThePapiok) [reply src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java]: done

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java (hunk 4 above)
Comment (ThePapiok) [reply src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java]: something like i added ?

Comment (ThePapiok) [reply src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java]: something like what i added ?

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java (hunk 5 above)
Comment (ThePapiok) [reply src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java]: something like what i added ?

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java (hunk 6 above)
Comment (ThePapiok) [reply src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java]: something like what i added ?

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java (hunk 7 above)
Comment (ThePapiok) [reply src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java]: something like what i added ?

File: src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    String mail = "test@test.com";
+    User user = new User();
+    user.setMail(mail);
+    RefreshTokenDTO refreshTokenDTO = new RefreshTokenDTO();
+    refreshTokenDTO.setToken("token");
+    Mockito.when(jwtService.extractUserName("token")).thenReturn("test@test.com");
+    Mockito.when(userRepository.findByMail(mail)).thenReturn(Optional.of(user));
+    Mockito.when(jwtService.isTokenValid("token", user)).thenReturn(true);
+    Mockito.when(jwtService.generateToken(user)).thenReturn("newToken");
+    Assertions.assertNotNull(authenticationService.refreshToken(refreshTokenDTO));
```
[the remark this answers is not in the document]
Comment (ThePapiok) [reply src/test/java/umk/mat/pajda/ProjektZespolowy/Services/AuthenticationServiceImplTest.java]: something like what i added ?

**Review by kacper-dabrowski — DISMISSED** [review]

File: src/main/java/umk/mat/pajda/ProjektZespolowy/configs/SecurityConfig.java [hunk 9]
```diff
@@ -1,72 +1,83 @@
...
+                    .requestMatchers(
+                        HttpMethod.PATCH,
+                        "/user/editInformations",
+                        "/user/editPassword",
+                        "/user/editEmail")
+                    .authenticated()
+                    .requestMatchers(HttpMethod.DELETE, "/user")
+                    .authenticated()
+                    .requestMatchers(
+                        HttpMethod.GET,
```
Comment (DarthRevan77797) [inline src/main/java/umk/mat/pajda/ProjektZespolowy/configs/SecurityConfig.java:51]: there is no requestMatchers for  HttpMethod.POST .    Add  requestMatchers  with  HttpMethod.POST for /review endpoint.

**Review by DarthRevan77797 — DISMISSED** [review]

**Pull request merged**